cube/cube.cpp

```diff
@@ -1470,14 +1325,14 @@ void Demo::create_surface() {
 void Demo::init_vk_swapchain() {
     create_surface();
     // Iterate over each queue to learn whether it supports presenting:
-    std::unique_ptr<vk::Bool32[]> supportsPresent(new vk::Bool32[queue_family_count]);
-    for (uint32_t i = 0; i < queue_family_count; i++) {
+    std::vector<vk::Bool32> supportsPresent(queue_props.size());
+    for (uint32_t i = 0; i < supportsPresent.size(); i++) {
```

Comment:
Comparing different types, also below

Reply:
`static_cast<uint32_t>(size())` should prevent the size comparison warnings, but I also modified the function a bit, hope that looks a bit cleaner.